### ISSUE TITLE
使用 Menu 组件 在开发者模式下  终端报错

### ISSUE THREAD

**Issue by ok2fly** [ant-design/ant-design#22638]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/ok2fly/antd-comp-bug-reproduce.git](https://github.com/ok2fly/antd-comp-bug-reproduce.git)

### Steps to reproduce
*  进入项目根目录安装依赖
yarn install   
*  运行项目
yarn start
*  打开 chrome console

### What is expected?
修复 控制台报错

### What is actually happening?
index.js:1 Warning: Legacy context API has been detected within a strict-mode tree.

The old API will be supported in all 16.x releases, but applications using it should migrate to the new version.

Please update the following components: Connect(MenuItem), Connect(SubMenu), Connect(SubPopupMenu), Provider, SubMenu

Learn more about this warning here: https://fb.me/react-legacy-context
    in Provider (created by Menu)
    in Menu (created by Context.Consumer)
    in InternalMenu (created by Context.Consumer)
    in Menu (at App.js:11)
    in div (at App.js:10)
    in App (at src/index.js:9)
    in StrictMode (at src/index.js:8)
console.<computed> @ index.js:1
printWarning @ react-dom.development.js:88
error @ react-dom.development.js:60
(匿名) @ react-dom.development.js:11432
push../node_modules/react-dom/cjs/react-dom.development.js.ReactStrictModeWarnings.flushLegacyContextWarning @ react-dom.development.js:11418
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js:23109
commitRootImpl @ react-dom.development.js:22396
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
scheduleUpdateOnFiber @ react-dom.development.js:21188
updateContainer @ react-dom.development.js:24373
(匿名) @ react-dom.development.js:24758
unbatchedUpdates @ react-dom.development.js:21903
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
./src/index.js @ index.js:7
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
1 @ serviceWorker.js:141
__webpack_require__ @ bootstrap:784
checkDeferredModules @ bootstrap:45
webpackJsonpCallback @ bootstrap:32
(匿名) @ main.chunk.js:1
index.js:1 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of CSSMotion which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://fb.me/react-strict-mode-find-node
    in ul (created by DOMWrap)
    in DOMWrap (created by SubPopupMenu)
    in SubPopupMenu (created by Connect(SubPopupMenu))
    in Connect(SubPopupMenu) (created by CSSMotion)
    in CSSMotion (created by ForwardRef)
    in ForwardRef (created by SubMenu)
    in li (created by SubMenu)
    in SubMenu (created by Connect(SubMenu))
    in Connect(SubMenu) (created by Context.Consumer)
    in SubMenu (at App.js:18)
    in ul (created by DOMWrap)
    in DOMWrap (created by SubPopupMenu)
    in SubPopupMenu (created by Connect(SubPopupMenu))
    in Connect(SubPopupMenu) (created by Menu)
    in Provider (created by Menu)
    in Menu (created by Context.Consumer)
    in InternalMenu (created by Context.Consumer)
    in Menu (at App.js:11)
    in div (at App.js:10)
    in App (at src/index.js:9)
    in StrictMode (at src/index.js:8)

| Environment | Info |
|---|---|
| antd | 4.0.4 |
| React | v16.13.1 |
| System | mac |
| Browser | chrome |

---
在开发环境下 页面显示正常，
控制台提示 报错

**Comment by zombieJ** [comment]
Need update `mini-store`. 
ref: https://github.com/yesmeck/mini-store/blob/master/src/Provider.js

**Comment by ok2fly** [comment]
> https://github.com/yesmeck/mini-store/blob/master/src/Provider.js

This warning is caused by the introduction of antd Menu component,
need i modify the Menu component source to fix it ?

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/9870

**Comment by ok2fly** [comment]
按需加载在开发这模式下会在控制台报错，这是不应该出现的啊， 该怎么解决？

**Comment by ok2fly** [comment]
> Need update `mini-store`.
> ref: https://github.com/yesmeck/mini-store/blob/master/src/Provider.js

应该怎么更新啊  ？  求助，

**Comment by ok2fly** [comment]
#22604  里面有人回答了 谢谢 @helloCheng09

删除index.js的React.StrictMode标签就可以了

**Comment by helloCheng09** [comment]
> #22604 里面有人回答了 谢谢 @helloCheng09
> 
> 删除index.js的React.StrictMode标签就可以了

thank for DaLao, @afc163

**Comment by ngosang** [comment]
This bug is not fixed in antd@4.1.2

```
Warning: Legacy context API has been detected within a strict-mode tree.

The old API will be supported in all 16.x releases, but applications using it should migrate to the new version.

Please update the following components: SubMenu

Learn more about this warning here: https://fb.me/react-legacy-context
    in SubMenu (at Sidebar.tsx:47)
    in ul (created by DOMWrap)
    in DOMWrap (created by SubPopupMenu)
    in SubPopupMenu (created by Connect(SubPopupMenu))
    in Connect(SubPopupMenu) (created by Menu)
    in Provider (created by Menu)
    in Menu (created by Context.Consumer)
    in InternalMenu (created by Context.Consumer)
    in Menu (at Sidebar.tsx:46)
    in div (created by Context.Consumer)
    in aside (created by Context.Consumer)
    in InternalSider (created by Context.Consumer)
    in Sider (at Sidebar.tsx:41)
    in Sidebar (created by ConnectFunction)
    in ConnectFunction (at AppLayout.tsx:15)
    in section (created by Context.Consumer)
    in BasicLayout (created by Context.Consumer)
    in Layout (at AppLayout.tsx:14)
    in Router (created by HashRouter)
    in HashRouter (at AppLayout.tsx:13)
    in AppLayout (at App.tsx:51)
    in App (created by ConnectFunction)
    in ConnectFunction (at src/index.tsx:15)
    in Provider (at src/index.tsx:14)
    in StrictMode (at src/index.tsx:13) index.js:1
    e index.js:1
    printWarning 0.chunk.js:186761
    error 0.chunk.js:186733
    flushLegacyContextWarning 0.chunk.js:198145
    flushLegacyContextWarning 0.chunk.js:198132
    flushRenderPhaseStrictModeWarningsInDEV 0.chunk.js:209633
    commitRootImpl 0.chunk.js:208930
    unstable_runWithPriority 0.chunk.js:223016
    runWithPriority$1 0.chunk.js:197748
    commitRoot 0.chunk.js:208915
    finishSyncRender 0.chunk.js:208332
    performSyncWorkOnRoot 0.chunk.js:208318
    flushSyncCallbackQueueImpl 0.chunk.js:197802
    unstable_runWithPriority 0.chunk.js:223016
    runWithPriority$1 0.chunk.js:197748
    flushSyncCallbackQueueImpl 0.chunk.js:197797
    flushSyncCallbackQueue 0.chunk.js:197785
    batchedUpdates$1 0.chunk.js:208388
    notify 0.chunk.js:215229
    notifyNestedSubs 0.chunk.js:215300
    handleChangeWrapper 0.chunk.js:215305
    dispatch 0.chunk.js:219462
    S 0.chunk.js:219038
    createThunkMiddleware 0.chunk.js:219213
    dispatch 0.chunk.js:219880
    fetchIndexersConfig main.chunk.js:2129
```

**Comment by azhen98** [comment]
感觉删除了严格模式是不是太过了,也没有其他好的办法
